src/unix/linux_like/android/mod.rs

```diff
@@ -528,16 +526,16 @@ s_no_extra_traits! {
     }
 
     pub struct dirent {
-        pub d_ino: u64,
-        pub d_off: i64,
+        pub d_ino: crate::ino_t,
```

Comment:
This is not correct, `ino_t` is [only used when `LP64`](https://cs.android.com/android/platform/superproject/main/+/main:bionic/libc/include/dirent.h;l=61-69?q=d_ino&ss=android%2Fplatform%2Fsuperproject%2Fmain:bionic%2Flibc%2Finclude%2F&start=1) is enabled (int is 32-bit, long int is 64-bit, pointers are 64-bit).

In all other cases (mostly 32-bit CPUs), this is an *explicit* `u64`, not an `ino_t`.

We could split `dirent` by architecture if we really wanted to use `ino_t` here, but on LP64 platforms, `ino_t` happens to be `u64` which is why this was working.